src/overlays/actors/ovl_En_Yabusame_Mark/z_en_yabusame_mark.c

```diff
@@ -40,17 +29,221 @@ static ColliderQuadInit D_80B434F0 = {
         ELEMTYPE_UNK0,
         { 0x00000000, 0x00, 0x00 },
         { 0x0001F824, 0x00, 0x00 },
-        TOUCH_NONE,
+        TOUCH_NONE | TOUCH_SFX_NORMAL,
         BUMP_ON,
         OCELEM_NONE,
     },
     { { { 0.0f, 0.0f, 0.0f }, { 0.0f, 0.0f, 0.0f }, { 0.0f, 0.0f, 0.0f }, { 0.0f, 0.0f, 0.0f } } },
 };
-*/
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Yabusame_Mark/EnYabusameMark_Destroy.s")
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Yabusame_Mark/EnYabusameMark_Init.s")
+const ActorInit En_Yabusame_Mark_InitVars = {
+    ACTOR_EN_YABUSAME_MARK,
+    ACTORCAT_PROP,
+    FLAGS,
+    OBJECT_GAMEPLAY_KEEP,
+    sizeof(EnYabusameMark),
+    (ActorFunc)EnYabusameMark_Init,
+    (ActorFunc)EnYabusameMark_Destroy,
+    (ActorFunc)EnYabusameMark_Update,
+    NULL,
+};
+
+Vec3f collisionVerticies[] = {
+    { 70.0f, 70.0f, 0.0f },      { 70.0f, -70.0f, 0.0f },      { -70.0f, 70.0f, 0.0f },    { -70.0f, -70.0f, 0.0f },
+    { 90.0f, 130.0f, -120.0f },  { -25.0f, -80.0f, -130.0f },  { 90.0f, 130.0f, 120.0f },  { -25.0f, -80.0, 130.0f },
+    { 115.0f, 160.0f, -150.0f }, { -50.0f, -140.0f, -160.0f }, { 115.0f, 160.0f, 150.0f }, { -50.0f, -140.0f, 160.0f },
+};
+
+#ifdef NON_MATCHING
+Vec3f D_80B435F0[] = {
+    { 3382.0f, 1734.0f, -4946.0f }, // small, furthest from entrance
+    { 3360.0f, 1734.0f, 495.0f },   // small, closest to entrance
+    { 4517.0f, 1682.0f, -1779.0f }, // medium, on the right
+    { 4517.0f, 1682.0f, -2813.0f }, // medium, on the left
+    { 4522.0f, 1727.0f, -2296.0f }, // large in the center
+};
 
+typedef struct {
+    /* 0x00 */ f32 unk_0; // first ring
+    /* 0x04 */ f32 unk_4; // second ring
+    /* 0x08 */ f32 unk_8; // outside edge
+    /* 0x0C */ f32 unk_C;
+} unknownStruct; // size = 0x10
+
+unknownStruct D_80B4362C[] = {
+    { 20.0f, 40.0f, 60.0f, 777.0f },
+    { 40.0f, 80.0f, 120.0f, 777.0f },
+    { 40.0f, 120.0f, 160.0f, 777.0f },
+};
+#else
+s32 D_80B435F0 = 0x45536000;
+s32 D_80B435F4 = 0x44D8C000;
+s32 D_80B435F8[] = {
+    0xC59A9000, 0x45520000, 0x44D8C000, 0x43F78000, 0x458D2800, 0x44D24000, 0xC4DE6000,
+    0x458D2800, 0x44D24000, 0xC52FD000, 0x458D5000, 0x44D7E000, 0xC50F8000,
+};
+s32 D_80B4362C[] = {
+    0x41A00000, 0x42200000, 0x42700000, 0x44424000, 0x42200000, 0x42A00000,
+    0x42F00000, 0x44424000, 0x42200000, 0x42F00000, 0x43200000, 0x44424000,
+};
```

Comment:
I assume this data is in a `NON_MATCHING` block because the asm function references fake symbols like `D_80B435F4` and `D_80B435F8`, but you shouldn't actually need to do this. I think you could replace the references to `D_80B435F4` by `D_80B435F0+4` and `D_80B435F8` by `D_80B435F0+8` in the asm, so you no longer need a `NON_MATCHING` block for the data.